File: templates/_composition/Prism/Feature.WhatsNewPrompt/Services/WhatsNewDisplayService.cs
using Param_ItemNamespace.Views;
using Param_ItemNamespace.Helpers;
using System;
using System.Threading.Tasks;
using Windows.ApplicationModel;

namespace Param_ItemNamespace.Services
{
    // For instructions on testing this service see https://github.com/Microsoft/WindowsTemplateStudio/tree/master/docs/features/whats-new-prompt.md
    public class WhatsNewDisplayService : IWhatsNewDisplayService
    {
        public async Task ShowIfAppropriateAsync()
        {
            var currentVersion = PackageVersionToReadableString(Package.Current.Id.Version);

            var lastVersion = await Windows.Storage.ApplicationData.Current.LocalSettings.ReadAsync<string>(nameof(currentVersion));

            if (lastVersion == null)
            {
                await Windows.Storage.ApplicationData.Current.LocalSettings.SaveAsync(nameof(currentVersion), currentVersion);
            }
            else
            {
                if (currentVersion != lastVersion)
                {
                    await Windows.Storage.ApplicationData.Current.LocalSettings.SaveAsync(nameof(currentVersion), currentVersion);

                    var dialog = new WhatsNewDialog();
                    await dialog.ShowAsync();
                }
            }
        }

        private static string PackageVersionToReadableString(PackageVersion packageVersion)
        {
            return $"{packageVersion.Major}.{packageVersion.Minor}.{packageVersion.Build}.{packageVersion.Revision}";
        }
    }
}

File: code/TemplateStudioForUWP/Templates/Ft/WhatsNewPrompt._VB/Services/WhatsNewDisplayService.vb
Imports Param_RootNamespace.Views

Imports Microsoft.Toolkit.Uwp.Helpers

Imports Windows.ApplicationModel.Core
Imports Windows.UI.Core

Namespace Services
    ' For instructions on testing this service see https://github.com/microsoft/TemplateStudio/blob/main/docs/UWP/features/whats-new-prompt.md
    Public Module WhatsNewDisplayService
        Dim shown As Boolean = False

        Friend Async Function ShowIfAppropriateAsync() As Task
            Await CoreApplication.MainView.CoreWindow.Dispatcher.RunAsync(CoreDispatcherPriority.Normal,
            Async Sub()
                If SystemInformation.Instance.IsAppUpdated AndAlso Not shown Then
                    shown = True
                    Dim dialog = New WhatsNewDialog()
                    Await dialog.ShowAsync()
                End If
            End Sub)
        End Function
    End Module
End Namespace
